components/code_panel.js

```diff
@@ -1,35 +1,87 @@
 let {div, header, footer, button, p, hJSX} = require('@cycle/dom')
 import isolate from '@cycle/isolate';
 
+let tommy = require('tommy-the-runner')
+let Rx = require('rx')
+
 function intent({ DOM, context }) {
     let buttonClicks$ = DOM.select('.submit-button').events('click')
-        .map(ev => {
-            return new Date().getTime()
+        .flatMap(ev => {
+           var userCode = document.getElementById('user-code').value
+
+            // TODO: retrieve specs from the original JSON (store in context?)
+            var specs = `
+                            const expect = require('chai').expect
+                            const sum = require('subject')
+                            describe('sum', function () {
+                                it('should sum two numbers', function () {
+                                    expect(sum(1, 1)).to.equal(2)
+                                })
+                            })
+                        `
+
+            return Rx.Observable.fromPromise(tommy.run(userCode, specs)
+                .then(reporter => {
+                    var stats = reporter.stats
+
+                    var failed = stats.failures
+                    var passed = stats.passes
+                    var pending = stats.pending
+
+                    var numTests = stats.tests
+
+                    var result = { failed, passed, pending, numTests, status: 'pending' }
+
+                    if (failed > 0) {
+                        result.status = 'failed'
+                    }
+
+                    if (passed == numTests) {
+                        result.status = 'passed'
+                    }
+
+                    return result
+                })
+                .catch(err => {
+                    return { error: `Execution error: ${err}}` }
+                }))
         })
 
     return context
         .map(ctx => ctx.timestamp)
         .concat(buttonClicks$)
 }
 
-function view(timestamp$) {
-    return timestamp$.map(ts => {
-        return (
-            <div id="code">
-                <header>
-                    <div className="container-header">
-                        Your code
+function view(testResults$) {
+    return testResults$
+        .map(res => {
+            var resultsStatusClass = res.status
```

Comment:
Instead of conditionally applying 'hidden' class name you could apply state class name and leave an appearance for css

Html
```html
<div class="status-passed">
    <span class="error-message"></span>
</div>
```

Css
```css
.status-passed .error-message {
    display: none
}
```